### DISCUSSION TITLE
Feature-level analysis - Mean abundance

### DISCUSSION THREAD

**Discussion by KidneyNode** [cafferychen777/MicrobiomeStat#66] [Q&A] [answered]
Hi, just a question regarding how "mean abundance" is calculated, for example in "generate_taxa_test_pair".

Firstly, is it the mean abundance of all samples at across every timepoint, or at a specific timepoint (if time is a variable"? And is it calculated as "relative abundance" or raw counts or is there a transformation ie. log2/clr/css?

**Comment by cafferychen777** [comment, marked as answer]
Hi @KidneyNode,

Thank you for your question about the calculation of "mean abundance" in the `generate_taxa_test_pair` function. I've reviewed the code and can provide some clarity:

1. Timepoint consideration:
   The mean abundance is calculated across all samples, regardless of timepoint. It's not specific to any particular timepoint, even if time is a variable in the analysis.

2. Calculation method:
   The calculation depends on the `feature.dat.type` parameter:
   - For "count" data: The raw counts are first normalized to relative abundances using Total Sum Scaling (TSS) before calculating the mean.
   - For "proportion" data: The mean is calculated directly from the input proportions.
   - For "other" data: The mean is calculated using the input values as-is, without any transformation.

3. Transformations:
   The mean abundance calculation does not involve log2, CLR, or CSS transformations. It represents the average relative abundance across all samples.

The specific line calculating mean abundance is:

```R
avg_abundance = mean(Freq)
```

Where `Freq` represents either the normalized counts (for "count" data) or the original proportions (for "proportion" data).

It's important to note that while this mean abundance is used for filtering and reporting, the actual differential abundance analysis in the `linda` function uses more sophisticated methods, including potential log transformations and compositional data analysis techniques.

Let me know if you need any further clarification!

Best,
Chen Yang

**Reply by cafferychen777** [reply]
@KidneyNode  I'd like to add an important point:

It's crucial to note that the calculation of "mean abundance" may vary across different functions in the MicrobiomeStat package. The method I described is specific to the `generate_taxa_test_pair` function, but other functions might use different approaches.

For instance:
- Some functions might calculate mean abundance at specific timepoints.
- Others might use different normalization methods before calculating the mean.
- Some might apply transformations like log2, CLR, or CSS before or after calculating the mean.

Therefore, it's always best to check the specific implementation in each function you're working with. The calculation can be function-specific and may depend on the particular analysis goals of that function.

If you're unsure about how mean abundance is calculated in a specific function, feel free to ask about that particular case, and we can look into the details for that function.

This "case-by-case" approach ensures you're using the correct interpretation of mean abundance for each analysis you perform with MicrobiomeStat.
